pkg/handler/offchain_tip_bot_test.go

```diff
@@ -0,0 +1,244 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/defipod/mochi/pkg/config"
+	"github.com/defipod/mochi/pkg/entities"
+	"github.com/defipod/mochi/pkg/logger"
+	"github.com/defipod/mochi/pkg/repo/pg"
+	"github.com/defipod/mochi/pkg/request"
+	"github.com/defipod/mochi/pkg/service"
+	mock_coingecko "github.com/defipod/mochi/pkg/service/coingecko/mocks"
+	"github.com/defipod/mochi/pkg/util/testhelper"
+	"github.com/gin-gonic/gin"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandler_GetUserBalances(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	cfg := config.LoadTestConfig()
+	log := logger.NewLogrusLogger()
+	svc, _ := service.NewService(cfg, log)
+	coingeckoMock := mock_coingecko.NewMockService(ctrl)
+	svc.CoinGecko = coingeckoMock
+
+	entity := entities.New(cfg, log, repo, nil, nil, nil, nil, svc, nil, nil, nil)
+	h := &Handler{
+		entities: entity,
+		log:      log,
+	}
+
+	tests := []struct {
+		name             string
+		query            string
+		wantCode         int
+		wantResponsePath string
+	}{
```

Comment:
need add case user not have any balance (dont have any record in db)

Reply:
work the same as case dont have balances